src/www/scripts/app.js

```diff
@@ -622,26 +638,48 @@ module.exports = (function() {
 
             setupDirectoryLocations();
 
+            var tokenStore = requirePin ? new TokenStore(secureStore) : new TokenStore(localStore);
```

Comment:
In case of a failed automatic login, we want to make sure that all previous sessions / tokens are removed, so that the user is forced to login in manually again. With this line, we make sure that the correct tokenStore is used.